cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -332,17 +332,37 @@ func parseKubeReserved(kubeReserved string) (apiv1.ResourceList, error) {
 }
 
 func extractLabelsFromKubeEnv(kubeEnv string) (map[string]string, error) {
-	labels, err := extractFromKubeEnv(kubeEnv, "NODE_LABELS")
+	// In v1.10+, labels are only exposed for the autoscaler via AUTOSCALER_ENV_VARS
+	// see kubernetes/kubernetes#61119. We try AUTOSCALER_ENV_VARS first, then
+	// fall back to the old way.
+	labels, err := extractAutoscalerVarFromKubeEnv(kubeEnv, "node_labels")
 	if err != nil {
-		return nil, err
+		glog.Infof("node_labels not found via AUTOSCALER_ENV_VARS, will try NODE_LABELS")
+		labels, err := extractFromKubeEnv(kubeEnv, "NODE_LABELS")
+		if err != nil {
+			return nil, err
+		}
+		return parseKeyValueListToMap(labels)
 	}
 	return parseKeyValueListToMap(labels)
 }
 
 func extractTaintsFromKubeEnv(kubeEnv string) ([]apiv1.Taint, error) {
-	taints, err := extractFromKubeEnv(kubeEnv, "NODE_TAINTS")
+	// In v1.10+, taints are only exposed for the autoscaler via AUTOSCALER_ENV_VARS
+	// see kubernetes/kubernetes#61119. We try AUTOSCALER_ENV_VARS first, then
+	// fall back to the old way.
+	taints, err := extractAutoscalerVarFromKubeEnv(kubeEnv, "node_taints")
 	if err != nil {
-		return nil, err
+		glog.Infof("node_taints not found via AUTOSCALER_ENV_VARS, will try NODE_TAINTS")
+		taints, err := extractFromKubeEnv(kubeEnv, "NODE_TAINTS")
+		if err != nil {
+			return nil, err
+		}
+		taintMap, err := parseKeyValueListToMap(taints)
```

Comment:
Same comment as above, but in this case there is more duplicated code. Once you get the taints extracted, you can leave the if block and parse them into a map in one place.